internal/agent/agent.go

```diff
@@ -30,19 +28,15 @@ var configPath string
 type Agent struct {
 	initialized bool
 	config      *Config
-	redis       cache.RedisCache
-	State       *State
+	redis       state.RedisCache
+	State       *state.State
 	auth        Authenticator
-	vlib        vbox.Library
+	vlib        *virtual.VboxLibrary
```

Comment:
Not sure why you made this change in the virtual

Reply:
Removed the interfaces which made no sense to have

Reply:
Also gathered docker and vbox under the virtual package.
May make sense to split them into their own packages again for readability